airflow/providers/google/cloud/transfers/azure_fileshare_to_gcs.py

```diff
@@ -0,0 +1,191 @@
+    :type dest_gcs: str
+    :param delegate_to: Google account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param replace: Whether you want to replace existing destination files
+        or not.
+    :type replace: bool
+    :param gzip: Option to compress file for upload
+    :type gzip: bool
+    :param google_impersonation_chain: Optional Google service account to impersonate using
+        short-term credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type google_impersonation_chain: Optional[Union[str, Sequence[str]]]
+
+    Note that ``share_name``, ``directory_name``, ``prefix``, ``delimiter`` and ``dest_gcs`` are
+    templated, so you can use variables in them if you wish.
+    """
+
+    template_fields: Iterable[str] = (
+        'share_name',
+        'directory_name',
+        'prefix',
+        'dest_gcs',
+        # 'google_impersonation_chain',
+    )
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        share_name: str,
+        directory_name: Optional[str] = None,
+        prefix: str = '',
+        wasb_conn_id: str = 'wasb_default',
+        gcp_conn_id: str = 'google_cloud_default',
+        dest_gcs: str,
+        delegate_to: Optional[str] = None,
+        replace: bool = False,
+        gzip: bool = False,
+        google_impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.share_name = share_name
+        self.directory_name = directory_name
+        self.prefix = prefix
+        self.wasb_conn_id = wasb_conn_id
+        self.gcp_conn_id = gcp_conn_id
+        self.dest_gcs = dest_gcs
+        self.delegate_to = delegate_to
+        self.replace = replace
+        self.gzip = gzip
+        self.google_impersonation_chain = google_impersonation_chain
+
+        if dest_gcs and not self._gcs_object_is_directory(self.dest_gcs):
+            self.log.info(
+                'Destination Google Cloud Storage path is not a valid '
+                '"directory", define a path that ends with a slash "/" or '
+                'leave it empty for the root of the bucket.'
+            )
+            raise AirflowException(
+                'The destination Google Cloud Storage path must end with a slash "/" or be empty.'
```

Comment:
Hm, do we check here if the bucket is empty? If not then this message is slightly confusing

Reply:
I believe `_gcs_object_is_directory` function takes care of it - if I understand it correctly it returns `True` if bucket is empty so this exception won't be raised.